src/Schema.ts

```diff
@@ -806,15 +806,15 @@ export abstract class Schema {
                         //
                         // use cached bytes directly if is from Schema type.
                         //
-                        filteredBytes = filteredBytes.concat(changeTree.caches[fieldIndex]);
+                        filteredBytes.push.apply(filteredBytes, changeTree.caches[fieldIndex] || [undefined]);
```

Comment:
```suggestion
                        filteredBytes.push(...(changeTree.caches[fieldIndex] ?? []));
```

Reply:
You have to push `undefined` into the array otherwise unit tests fail.